src/components/Tree/Tree.tsx

```diff
@@ -0,0 +1,253 @@
+import { Base, DefaultSlot, Component, Prop } from '@/core';
+
+type Node = {
+  columns: any[];
+  children?: Node[] | [];
+};
+
+type NodeWithId = {
+  id: string;
+  columns: any[];
+  children?: NodeWithId[] | [];
+};
+
+type LinkNode = {
+  displayText?: string;
+  linkUrl: string;
+};
+
+export interface Props {
+  headers: string[];
+  treeData: Node[];
+}
+
+@Component('Tree')
+@DefaultSlot('Tree Content')
+export class Tree extends Base<Props> {
+  @Prop('headers of the tree', { type: Array, required: true })
+  public headers!: string[];
+
+  @Prop('data of the tree', { type: Array, required: true })
+  public treeData!: Node[];
+
+  public iStates: boolean[] = [];
```

Comment:
This is only exposed for testing purposes? From looking at the code my gut feeling is that testing this is not needed. In your test you already test `numberOfElements` which is basically the same?

Thinking more about it I would not test any data structure that is naturally private. If something is private it is private. :D Instead of checking `numberOfElements` we could count the number of rows rendered. Wouldn't that also work and this would more or less test something public because it is specified by the fundamentals "spec" and it is something the user actually sees.

Reply:
Yea, agree with you. I've changed back to private variables and check on the number of <li> for testing.